### PR TITLE
LPS-150565 Add configuration values

### PR DESCRIPTION
**Pull request by yuhai**
Resending https://github.com/liferay-core-infra/liferay-portal/pull/2462

@dantewang, for the last commit, one unrelated language key was removed by buildLang command. Is this good for the commit?

cc @vendeltoreki

### CODE REVIEW

**Comment by Kvale1733** [comment]
@yuhai No regressions found.

**Comment by yuhai** [comment]
@dantewang, I just resolved bnd file conflict. I will remove "QA passed" and trigger ci test again.

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

**Review by dantewang — APPROVED** [review]

**Comment by catarineaguiar** [comment]
@yuhai no regressions found.

File: modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/background/task/LayoutSetPrototypeImportBackgroundTaskExecutor.java [hunk 1]
```diff
@@ -184,6 +189,32 @@ public BackgroundTaskResult execute(BackgroundTask backgroundTask)
...
+	protected void activate(
+			BundleContext bundleContext, Map<String, Object> properties)
+		throws PortalException {
+
+		setBackgroundTaskStatusMessageTranslator(
+			new LayoutExportImportBackgroundTaskStatusMessageTranslator());
+
+		LayoutSetPrototypeSystemConfiguration
+			layoutSetPrototypeSystemConfiguration =
+				ConfigurableUtil.createConfigurable(
```
Comment (dantewang) [inline modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/background/task/LayoutSetPrototypeImportBackgroundTaskExecutor.java:189]: @yuhai This component has configuration, but it has no `@Modified` method.

See https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-modification

> If the modified attribute is not specified, then the component configuration will become unsatisfied if its component properties use a Configuration object and that Configuration object is modified in any way.

So we need a `@Modified` method.

File: modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java [hunk 2]
```diff
@@ -42,15 +46,25 @@
...
 public class BackgroundTaskMessagingConfigurator {
 
 	@Activate
-	protected void activate(BundleContext bundleContext) {
+	protected void activate(
+			BundleContext bundleContext, Map<String, Object> properties)
+		throws PortalException {
+
+		BackgroundTaskManagerConfiguration backgroundTaskManagerConfiguration =
+			ConfigurableUtil.createConfigurable(
```
Comment (dantewang) [inline modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java:58]: @yuhai This configuration also suffers from not being able to update when configuration is modified.

**Review by dantewang — CHANGES_REQUESTED** [review]
@yuhai @vendeltoreki Just notice a problem, see my comments in the code.

When configuration is modified, the SCR will invoke `@Modified` method without deactivating and re-activating the component.

Based on that, if you have any resource of which the lifecycle is managed by the `@Activate` and `@Deactivate` methods, like "creating and starting a thread pool in `activate()` and shutting down it in `deactivate()`", you can't just add `@Modified` to the `activate()` method like many components in Liferay do.

If the creation of the resource has nothing to do with the configuration, then you shouldn't re-create it in `@Modified`; if the creation of the resource is based on the configuration, then you must destroy the previous resource and re-create a new one using the updated configuration.

**Comment by dantewang** [comment]
Closing the PR for now, please resend after fixing the `@Modified` related problem.

**Pull request closed without merging**